File: src/main/java/nz/co/searchwellington/utils/ColumnSplitter.java
package nz.co.searchwellington.utils;

import java.math.BigDecimal;
import java.util.List;

import org.springframework.stereotype.Component;

@Component
public class ColumnSplitter <T extends Object> {
        
    public List<T> left(List<T> source) {
        BigDecimal size  = new BigDecimal(source.size());        
        int splitPoint = size.divide(new BigDecimal(2), BigDecimal.ROUND_UP).intValue();
        return source.subList(0, splitPoint);               
    }
    
    public List<T> right(List<T> source) {
        BigDecimal size  = new BigDecimal(source.size());        
        int splitPoint = size.divide(new BigDecimal(2), BigDecimal.ROUND_UP).intValue();
        return source.subList(splitPoint, source.size());               
    }

}

File: src/main/scala/nz/co/searchwellington/utils/ColumnSplitter.scala
package nz.co.searchwellington.utils

import java.math.BigDecimal
import java.util
import java.util.List

import org.springframework.stereotype.Component

@Component
class ColumnSplitter[T <: Object] {

  val NUMBER_OF_COLUMNS: BigDecimal = new BigDecimal(2)

  def left(source: List[T]): List[T] = {
    source.subList(0, splitPointFor(source))
  }

  def right(source: List[T]): List[T] = {
    source.subList(splitPointFor(source), source.size)
  }

  private def splitPointFor(source: util.List[T]): Int = {
    new BigDecimal(source.size).divide(NUMBER_OF_COLUMNS, BigDecimal.ROUND_UP).intValue
  }

}